src/search.cpp

```diff
@@ -1074,15 +1072,16 @@ Value Search::Worker::search(
                 int seeHist = std::clamp(captHist / 31, -137 * depth, 125 * depth);
                 if (!pos.see_ge(move, -158 * depth - seeHist))
                 {
-                    bool skip = true;
-                    if (depth > 2 && !capture && givesCheck && alpha < 0
+                    bool mayStalemateTrap = depth > 2 && givesCheck && alpha < 0
+                        && !capture // we consider that captures will likely destroy the stalemate configuration
```

Comment:
Actually !capture condition is there because while processing captures we are not sure that already all moves (quiets too) are generated in movepicker.
Maybe it is more correct to put the condition on movepickers stage...

Reply:
..just recognized that it's practically impossible that a Queen/Rook sacrifice is returned as a GOOD_CAPTURE,
so practically it is guaranteed that quiets were already generated too.
Under that light I believe the !capture condition (and the givesCheck) condition can be simplified away.
I will do some test.